pipeline/matching/generators/clustering.py

```diff
@@ -0,0 +1,86 @@
+import math
+from typing import Iterator
+
+from pipeline.matching.core import MatchGenerator
+from pipeline.types import Match, MatchingInput, MatchMetadata
+
+GENERATOR_CLUSTERING = "clusteringGenerator"
+
+# TODO: Add type annotations to auxilary functions
+def cosine_similarity(v1, v2):
```

Comment:
Try adding type annotations but if the pre-commit gives you too much hassle, we can fix this together.